runtime/src/iree/base/internal/dynamic_library_win32.c

```diff
@@ -319,6 +319,36 @@ iree_status_t iree_dynamic_library_lookup_symbol(
   return iree_ok_status();
 }
 
+iree_status_t iree_dynamic_library_get_symbol_path(
+    void* symbol, iree_string_builder_t* out_path) {
+  HMODULE hm = NULL;
+  if (GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS |
+                            GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+                        (LPCSTR)symbol, &hm) == 0) {
+    return iree_make_status(IREE_STATUS_NOT_FOUND);
+  }
+
+  char* path_buffer = NULL;
+  iree_host_size_t path_capacity = 0;
+  iree_host_size_t path_size = 0;
+  IREE_RETURN_IF_ERROR(iree_string_builder_reserve_for_append(
+      out_path, 64, &path_buffer, &path_capacity));
+  while (1) {
+    path_size = GetModuleFileNameA(hm, path_buffer, path_capacity);
+    if (path_size == 0) {
+      return iree_make_status(IREE_STATUS_NOT_FOUND);
+    }
+    if (path_size == path_capacity) {
```

Comment:
you also need to check GetLastError() == ERROR_INSUFFICIENT_BUFFER (otherwise this will loop forever if the buffer happens to be exactly the right size)